utils/check_content_style.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/python
+# check_content_style.py
+# Copyright (c) 2022 by tibetiroka
+#
+# Endless Sky is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program. If not, see <https://www.gnu.org/licenses/>.
+
+import re
+import glob
+
+# Script that checks for common text content formatting pitfalls not covered by spellcheck or other tests.
+
+# The number of errors found in all files
+errors = 0
+# The list of errors in the current file; elements are tuples of the line numbers and the error messages
+error_list = []
+# Regex-based style checks for check_regexes()
+regexes = {
+	# Matches any space that is preceeded only by tabs or " or ` characters,
+	# except if it immediately follows a " or ` character.
+	"^[\\t`\"]*(?<![`\"]) ": "indentations should use tabs only",
+	# Matches multiple space characters.
+	"  ": "multiple consecutive space characters",
+	"\\s$": "trailing whitespace character"
+}
+# Precompile regexes
+regexes = {re.compile(regex): description for regex, description in regexes.items()}
+
+
+# Checks the format of all txt files.
+def check_content_style():
+	files = glob.glob('data/**/*.txt', recursive=True)
+	files.sort()
+	for file in files:
+
+		f = open(file, "r")
+		lines = f.readlines()
+		lines = [line.removesuffix('\n') for line in lines]
```

Comment:
Why remove the line ending?

This script currently uses the "universal newlines" mode of `open`, so CRLF will get translated to LF even if the host platform is not Windows. That seems like a scenario we would want to catch, and may be catchable by setting `newline=''` in the call to `open` and then comparing against `os.linesep`

Another scenario we'd probably want to catch would be the inclusion of the UTF BOM (byte-order mark), but you'd likely need to open the files in binary mode for that.

Reply:
Oh that's an interesting take. I removed line endings because they messed with the regexes, but I suppose we could include additional tests for checking the line separators.